### PR TITLE
Add support for React 16

### PR DESCRIPTION
**Pull request by philipp-spiess**
This updates the tests to use React 16 and demonstrates the issue described in #93.

```
$ mocha
/Users/philipp/dev/skin-deep/skin-deep.js:40
    shallowRenderer._instance._instance
                             ^

TypeError: Cannot read property '_instance' of null
    at Object.shallowRender (/Users/philipp/dev/skin-deep/skin-deep.js:40:30)
    at Suite.<anonymous> (/Users/philipp/dev/skin-deep/test/new.js:75:21)
    at context.describe.context.context (/Users/philipp/dev/skin-deep/node_modules/mocha/lib/interfaces/bdd.js:47:10)
    at Suite.<anonymous> (/Users/philipp/dev/skin-deep/test/new.js:71:5)
    at context.describe.context.context (/Users/philipp/dev/skin-deep/node_modules/mocha/lib/interfaces/bdd.js:47:10)
    at Suite.<anonymous> (/Users/philipp/dev/skin-deep/test/new.js:32:3)
    at context.describe.context.context (/Users/philipp/dev/skin-deep/node_modules/mocha/lib/interfaces/bdd.js:47:10)
    at Object.<anonymous> (/Users/philipp/dev/skin-deep/test/new.js:27:1)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at /Users/philipp/dev/skin-deep/node_modules/mocha/lib/mocha.js:220:27
    at Array.forEach (native)
    at Mocha.loadFiles (/Users/philipp/dev/skin-deep/node_modules/mocha/lib/mocha.js:217:14)
    at Mocha.run (/Users/philipp/dev/skin-deep/node_modules/mocha/lib/mocha.js:469:10)
    at Object.<anonymous> (/Users/philipp/dev/skin-deep/node_modules/mocha/bin/_mocha:404:18)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.runMain (module.js:604:10)
    at run (bootstrap_node.js:394:7)
    at startup (bootstrap_node.js:149:9)
    at bootstrap_node.js:509:3
```

### CODE REVIEW

File: skin-deep.js [hunk 1]
```diff
@@ -37,7 +39,9 @@ function shallowRender(element, context) {
     },
     shallowRenderer,
     /* eslint-disable no-underscore-dangle */
-    shallowRenderer._instance._instance
+    Versions.ReactBefore16
+      ? shallowRenderer._instance._instance
+      : shallowRenderer._instance
```
Comment (philipp-spiess) [inline skin-deep.js]: This difference can also be found in enzyme:

https://github.com/airbnb/enzyme/blob/16f15e028dfb2935467c4f263698c5d3e45e21ce/packages/enzyme-adapter-react-15/src/ReactFifteenAdapter.js#L181

vs.

https://github.com/airbnb/enzyme/blob/16f15e028dfb2935467c4f263698c5d3e45e21ce/packages/enzyme-adapter-react-16/src/ReactSixteenAdapter.js#L229

**Comment by philipp-spiess** [comment]
@glenjamin Ok it seems that the fix described in #93 is enough to make the tests pass again. I've also found the same change in `enzyme`. Anything that I might have forgotten?

**Comment by glenjamin** [comment]
Great work, thanks!

**Comment by glenjamin** [comment]
Releasing as `v1.2.0` as soon as CI passes.

**Pull request merged**